click/core.py

```diff
@@ -697,7 +704,7 @@ def main(self, args=None, prog_name=None, complete_var=None,
                     rv = self.invoke(ctx)
                     if not standalone_mode:
                         return rv
-                    ctx.exit()
+                    ctx.exit(rv)
```

Comment:
Interesting that I get different testing results when using `cd tests && pytest` and `make test`. I can reproduce the test case failures on Travis using `make test`.